### PR TITLE
chore(core): fix isAscii() flag in DirectUtf8Sink

### PR DESCRIPTION
**Pull request by jerrinot**
The old code was pessimistic. It made the sink
to claim to be non-ascii whenever it received any CharSequence.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8Sink.java|1|1|100.00%|

File: core/src/main/java/io/questdb/std/str/DirectUtf8Sink.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public DirectUtf8Sink put(@Nullable Utf8Sequence us) {
 
     @Override
     public DirectUtf8Sink put(byte b) {
-        ascii = false;
+        ascii &= b >= 0;
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/std/str/DirectUtf8Sink.java:112]: i don't think this is correct, the idea was to never call put(byte) outside of non-ascii context. I would rather have assert there.

We should never be dealing with unknown bytes. If we know byte is ascii, we should call putAscii(byte) that exists

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/std/str/DirectUtf8Sink.java (hunk 1 above)
Comment (jerrinot) [reply core/src/main/java/io/questdb/std/str/DirectUtf8Sink.java]: @bluestreak01 wdyt now?

Comment (bluestreak01) [reply core/src/main/java/io/questdb/std/str/DirectUtf8Sink.java]: yep, this looks consistent with other utf8 sinks

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/bytes/DirectByteSink.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8Sink.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/bytes/DirectByteSink.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8Sink.java|3|3|100.00%|

**Comment by jerrinot** [comment]
the same problem is addressed by https://github.com/questdb/questdb/pull/4430

**Pull request closed without merging**